File: Java/Easy/Accepted/find-nearest-point-that-has-the-same-x-or-y-coordinate_2.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        
        int minDistance = Integer.MAX_VALUE;
        int minIndex = -1;
        for (int i = 0; i < points.length; i++) {
            int[] point = points[i];
            if (point[0] == x || point[1] == y) {
                int distance = Math.abs(point[0] - x) + Math.abs(point[1] - y);
                if (distance < minDistance) {
                    minDistance = distance;
                    minIndex = i;
                }
            }
        }
        return minIndex;
    }
}

File: Rust/Easy/Accepted/find-nearest-point-that-has-the-same-x-or-y-coordinate_2.rs
impl Solution {
    pub fn nearest_valid_point(x: i32, y: i32, points: Vec<Vec<i32>>) -> i32 {
        
        let mut ans = -1;
        let mut min = std::i32::MAX;
        for (i, p) in points.iter().enumerate() {
            if p[0] == x || p[1] == y {
                let d = (p[0] - x).abs() + (p[1] - y).abs();
                if d < min {
                    ans = i as i32;
                    min = d;
                }
            }
        }
        ans
    }
}